libr/util/str.c

```diff
@@ -1566,7 +1566,7 @@ R_API char *r_str_encoded_json(const char *buf, int buf_size, int encoding) {
 
 		const char *format = encoding == PJ_ENCODING_STR_ARRAY ? "%03u," : "%02X";
 		while (buf[loop] != '\0' && i < (new_sz - 1)) {
-			snprintf (encoded_str + i, sizeof (encoded_str), format, (ut8) buf[loop]);
+			sprintf (encoded_str + i, format, (ut8) buf[loop]);
```

Comment:
```suggestion
			snprintf (encoded_str + i,  new_sz - i, (ut8) buf[loop]);
```

Reply:
Why must snprintf? Here every time 2 or 4 chars plus a \0 is written, as we already calculated length no overflow happens. Checking boundary has overhead.

Reply:
all static code analysis tool report using unsafe C functions as an issue. even if it's safe because you know it wont overflow, its never a good idea to use them

Reply:
Sounds reasonable.